### DISCUSSION TITLE
Using ophys.ImagingPlane vs NWBFile.create_imaging_plane()

### DISCUSSION THREAD

**Discussion by jmdelahanty** [NeurodataWithoutBorders/helpdesk#10] [Q&A] [answered]
Hello everyone!

I was wondering about which of these two options I should be using when creating my NWB File for 2P experiments.

How should I best use create_imaging_plane and is there any difference between it and the ophys.ImagingPlane method? If I build an imaging plane with the NWBFile method, can I still append a TwoPhotonSeries to it when I have that ready?

Thanks for your help!

**Comment by bendichter** [comment, marked as answer]
`NWBFile.create_imaging_plane` creates an instance of an `ImagingPlane` and automatically adds it to the `NWBFile` object. I would suspect that is more convenient for you.

`TwoPhotonSeries` takes an `imaging_plane` argument, which links it to a given imaging plane. So all together, it's:

```python
nwbfile = NWBFile(...)
...
imaging_plane = nwbfile.create_imaging_plane(...)
...
nwbfile.add_acquisition(TwoPhotonSeries(imaging_plane=imaging_plane, ...))
```

**Reply by jmdelahanty** [reply]
Perfect! That's just what I was looking for. Thanks for your help Ben!
